Adaptors/S3/src/ObjectStorage.cs

```diff
@@ -48,16 +52,49 @@ public class ObjectStorage : IObjectStorage
   /// <param name="options">S3 object storage options</param>
   /// <param name="logger">Logger used to print logs</param>
   public ObjectStorage(AmazonS3Client         s3Client,
-                       string                 objectStorageName,
                        Options.S3             options,
                        ILogger<ObjectStorage> logger)
   {
     s3Client_          = s3Client;
-    objectStorageName_ = objectStorageName;
+    objectStorageName_ = "objectStorageName";
     bucketName_        = options.BucketName;
     logger_            = logger;
   }
 
+  /// <inheritdoc />
+  public async Task Init(CancellationToken cancellationToken)
+  {
+    if (!isInitialized_)
+    {
+      await AmazonS3Util.DoesS3BucketExistV2Async(s3Client_,
+                                                  bucketName_);
+    }
+
+    logger_.LogInformation("ObjectStorageFactory has correctly been initialized.");
```

Comment:
In the if ?

Reply:
I did not change the implementation of the Init. I took the one from the IObjectStorageFactory